File: www/node/test/p145.py
import math as math # 数値計算ライブラリ
import matplotlib.pyplot as plt # グラフ描画ライブラリ
dt = 0.01 # 微小時間 ( 時間間隔 )
v0 = 30 # 初速度
g = 9.8 # 重力加速度
x = [0] # 水平位置の初期値 0
y = [0] # 鉛直位置の初期値は 0
angle = 45.0 * math.pi / 180.0 # 投げ上げ角度
vx = [v0*math.cos(angle)] # 水平方向の初速度
vy = [v0*math.sin(angle)] # 鉛直方向の初速度
for i in range(1000):
 vx.append(vx[i]) # 微小時間後の水平方向の速度
 vy.append(vy[i]-g*dt) # 微小時間後の鉛直方向の速度
 x.append(x[i]+vx[i]*dt) # 微小時間後の水平位置
 y.append(y[i]+(vy[i]+vy[i+1])/2.0*dt) # 微小時間後の鉛直位置
 if y[i] < 0 : # もし鉛直位置が 0 を下回ったら
  break # ループ中断
plt.plot(x,y) # 位置の配列をプロット
plt.title("parabollic motion") # グラフのタイトル
plt.xlabel("distance") #x 軸ラベル
plt.ylabel("height") #y 軸ラベル
plt.show()

File: www/node/test/js/p145.js
define('__main__',function (require,exports,module) {
  var PYLIB=require('http://localhost/runtime/lib/python/PyLib.js');
  var range=PYLIB.range;
  var input=PYLIB.input;
  var str=PYLIB.str;
  var int=PYLIB.int;
  var float=PYLIB.float;
  var len=PYLIB.len;
  var type=PYLIB.type;
  var quit=PYLIB.quit;
  var exit=PYLIB.exit;
  var sorted=PYLIB.sorted;
  var fillRect=PYLIB.fillRect;
  var setColor=PYLIB.setColor;
  var setTimeout=PYLIB.setTimeout;
  var clearRect=PYLIB.clearRect;
  var clear=PYLIB.clear;
  var open=PYLIB.open;
  PYLIB.LoopChecker.reset();
  var math=require('http://localhost/runtime/lib/python/py_math.js').install(PYLIB);
  var plt=require('http://localhost/runtime/lib/python/py_matplotlib,pyplot.js').install(PYLIB);
  var dt=0.01;
  var v0=30;
  var g=9.8;
  var x=[0];
  var y=[0];
  var angle=PYLIB.wrap(PYLIB.wrap(45.0).__mul__(math.pi)).__div__(180.0);
  var vx=[PYLIB.wrap(v0).__mul__(math.cos(angle))];
  var vy=[PYLIB.wrap(v0).__mul__(math.sin(angle))];
  var i;
  for (i of range(1000)) {
    PYLIB.LoopChecker.check();
    vx.append(vx[i]);
    vy.append(PYLIB.wrap(vy[i]).__sub__(PYLIB.wrap(g).__mul__(dt)));
    x.append(PYLIB.wrap(x[i]).__add__(PYLIB.wrap(vx[i]).__mul__(dt)));
    y.append(PYLIB.wrap(y[i]).__add__(PYLIB.wrap(PYLIB.wrap((PYLIB.wrap(vy[i]).__add__(vy[PYLIB.wrap(i).__add__(1)]))).__div__(2.0)).__mul__(dt)));
    if (PYLIB.wrap(y[i]).__lt__(0)) {
      PYLIB.LoopChecker.check();
      break;
    }
  }plt.plot(x,y);
  plt.title("parabollic motion");
  plt.xlabel("distance");
  plt.ylabel("height");
  plt.show();
});
requirejs(['__main__'],function(){});
